modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java

```diff
@@ -8111,6 +8111,15 @@ private void _removeDDMFormFieldValues(
 		}
 	}
 
+	private String _removeTrailingSlashes(String friendlyURL) {
+		if (friendlyURL.endsWith(StringPool.SLASH)) {
+			return _removeTrailingSlashes(
+				friendlyURL.substring(0, friendlyURL.length() - 1));
+		}
+
+		return friendlyURL;
+	}
```

Comment:
Can you avoid the recursive call? Ideally, you should also try to avoid doing multiple calls to `substring`. It would be better to look for the last character that is not a slash (with a loop, not recursively), and then do the substring once.

Reply:
thanks!

I will change this

Reply:
Do you think that extract the new function (`_findLastNoncharPosition`) is worth to extract to StringUtils ??